exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/ParquetTableMetadataProviderImpl.java

```diff
@@ -126,22 +128,23 @@ public Path getSelectionRoot() {
    * @return list of cache files found in the given directory path
    */
   public List<Path> populateMetaPaths(Path p, DrillFileSystem fs) throws IOException {
-    List<Path> metaFilepaths = new ArrayList<>();
-    for (String filename : Metadata.CURRENT_METADATA_FILENAMES) {
-      metaFilepaths.add(new Path(p, filename));
-    }
-    for (String filename : Metadata.OLD_METADATA_FILENAMES) {
-      // Read the older version of metadata file if the current version of metadata cache files donot exist.
+    if (fs.isDirectory(p)) {
+      List<Path> metaFilepaths = Arrays.stream(Metadata.CURRENT_METADATA_FILENAMES)
+          .map(filename -> new Path(p, filename))
+          .collect(Collectors.toList());
+      for (String filename : Metadata.OLD_METADATA_FILENAMES) {
+        // Read the older version of metadata file if the current version of metadata cache files do not exist.
+        if (fileExists(fs, metaFilepaths)) {
+          return metaFilepaths;
+        }
+        metaFilepaths.clear();
+        metaFilepaths.add(new Path(p, filename));
```

Comment:
Does this mean that for the last iteration we’ll add file name to metaFilepaths but will never check if it exists and return empty collection?

Reply:
No, we will check whether the metadata files from the last iteration exist below the loop and in this case return list with files or an empty collection if they do not exist.

Reply:
Ok, thanks. Missed check in the end.